### PR TITLE
[helm] Add support for TLS config blocks

### PR DESCRIPTION
**Pull request by wallyqs**
Example:

```sh
helm install nats -f ../deploy-nats-gateways.yaml ./charts/nats
```

where the yaml is:

```yaml
nats:
  externalAccess: true
  logging:
    debug: true
    trace: true
  tls:
    secret:
      name: nats-client-tls
    ca: "ca.crt"
    cert: "tls.crt"
    key: "tls.key"

cluster:
  enabled: true
  tls:
    secret:
      name: nats-server-tls
    ca: "ca.crt"
    cert: "tls.crt"
    key: "tls.key"

leafnodes:
  enabled: true

  tls:
    secret:
      name: nats-server-tls
    ca: "ca.crt"
    cert: "tls.crt"
    key: "tls.key"

  remotes:
    - url: tls://connect.ngs.global:7422
      credentials:
        secret:
          name: ngs-creds
          key: NGS.creds

gateway:
  enabled: true
  name: aws-useast2

  # Shared for all gateways
  tls:
    secret:
      name: nats-server-tls
    ca: "ca.crt"
    cert: "tls.crt"
    key: "tls.key"

# Add system credentials to the nats-box instance for example
natsbox:
  enabled: true

  credentials:
    secret:
      name: nats-sys-creds
      key: sys.creds
```

### CODE REVIEW

File: helm/charts/nats/templates/statefulset.yaml [hunk 1]
```diff
@@ -175,6 +205,32 @@ spec:
...
+          {{- with .Values.leafnodes.tls }}
+          {{ $secretName := .secret.name }}
+          - name: {{ $secretName }}-leafnodes-volume
+            mountPath: /etc/nats-certs/leafnodes/{{ $secretName }}
+          {{- end }}
+          {{- with .Values.gateway.tls }}
+          {{ $secretName := .secret.name }}
+          - name: {{ $secretName }}-gateways-volume
+            mountPath: /etc/nats-certs/gateways/{{ $secretName }}
+          {{- end }}
```
Comment (wallyqs) [inline helm/charts/nats/templates/statefulset.yaml:232]: These should also be mounted by the reloader sidecar so that it tracks the changes of the certs, but we can do that in a separate PR...

File: helm/charts/nats/values.yaml [hunk 2]
```diff
@@ -43,6 +43,18 @@ nats:
     connectErrorReports: 
     reconnectErrorReports: 
 
+  #######################
+  #                     #
+  #  TLS Configuration  #
+  #                     #
```
Comment (matthiashanel) [inline helm/charts/nats/values.yaml:49]: I assume this is a file ppl would look at?
Maybe in all of these tls sections add a linkt to https://docs.nats.io/nats-server/configuration/securing_nats/tls#problems-with-self-signed-certificates
As a warning what to watch out for.

File: helm/charts/nats/values.yaml [hunk 3]
```diff
@@ -56,17 +68,48 @@ leafnodes:
...
 # https://docs.nats.io/nats-server/configuration/gateways
 #
 gateway:
   enabled: false
   name: 'default'
-  # List of remote gateways
+
+  #############################
+  #                           #
+  #  List of remote gateways  #
```
Comment (matthiashanel) [inline helm/charts/nats/values.yaml:94]: there is a per gateway tls option as well. mention that this was explicitly left out?

File: helm/charts/nats/values.yaml [hunk 4]
```diff
@@ -56,17 +68,48 @@ leafnodes:
   # remotes:
   #   - url: "tls://connect.ngs.global:7422"
 
+  #######################
```
Comment (matthiashanel) [inline helm/charts/nats/values.yaml:71]: Same as for gateways. (per gateway tls section?)

Comment (matthiashanel) [reply helm/charts/nats/values.yaml:71]: I'm actually not quite sure if the outer tls section is the default in case per lleafnode is not provided

File: helm/charts/nats/values.yaml (hunk 3 above)
Comment (wallyqs) [reply helm/charts/nats/values.yaml:94]: explicitly left out for now for both leafnode remotes and gateway remotes to reduce size of PR a bit...

File: helm/charts/nats/values.yaml (hunk 2 above)
Comment (wallyqs) [reply helm/charts/nats/values.yaml:49]: that sounds good!

**Comment by ondrejtomcik** [comment]
Greetings @wallyqs  @matthiashanel 
This feature was merged but looks like a human error occurred during merge conflict resolution. This feature is not in the master nor in v0.3.0.

**Comment by wallyqs** [comment]
@ondrejtomcik you are right 😨 It looks like something may have happened with the merge probably during the Github outage...

**Comment by wallyqs** [comment]
@ondrejtomcik can you try `helm repo update` and use release v0.3.2? I think this should be fixed there.  Here is an example:

```yaml
# Authentication setup
auth:
  enabled: true

  # Reference to the Operator JWT which will be mounted as a volume,
  # shared with the account server in this case.
  operatorjwt:
    configMap:
      name: operator-jwt
      key: KO.jwt

  # Public key of the System Account
  systemAccount: AAITGVORQ4VHOQ32A7XMEKUIDMJ5GSAYSUGOV6GZAVSQWYFAL72DIXLC

  resolver:
    type: URL

    # 
    # NOTE: In case the account server URL is not present
    # in the operator JWT then need to set it explicitly here.
    # 
    url: "http://nats-account-server:9090/jwt/v1/accounts/"

nats:
  externalAccess: true
  logging:
    debug: true
    trace: true
  tls:
    secret:
      name: nats-client-tls
    ca: "ca.crt"
    cert: "tls.crt"
    key: "tls.key"

cluster:
  enabled: true
  tls:
    secret:
      name: nats-server-tls
    ca: "ca.crt"
    cert: "tls.crt"
    key: "tls.key"

leafnodes:
  enabled: true

  tls:
    secret:
      name: nats-server-tls
    ca: "ca.crt"
    cert: "tls.crt"
    key: "tls.key"

  remotes:
    - url: tls://connect.ngs.global:7422
      credentials:
        secret:
          name: ngs-creds
          key: NGS.creds

gateway:
  enabled: true
  name: aws-useast2

  # Shared for all gateways
  tls:
    secret:
      name: nats-server-tls
    ca: "ca.crt"
    cert: "tls.crt"
    key: "tls.key"

  # gateways:
  #   - name: euwest1
  #     url: tls://euwest1.aws.ngs.global:7522

# Add system credentials to the nats-box instance for example
natsbox:
  enabled: true

  credentials:
    secret:
      name: nats-sys-creds
      key: sys.creds
```
